pkg/deploy/controller/deployment/controller_test.go

```diff
@@ -524,3 +612,21 @@ func unrelatedPod(deployment *kapi.ReplicationController) *kapi.Pod {
 		},
 	}
 }
+
+func ttlNonZeroPod() *kapi.Pod {
+	ttl := int64(10)
+	return &kapi.Pod{
+		Spec: kapi.PodSpec{
+			ActiveDeadlineSeconds: &ttl,
+		},
+	}
+}
+
+func ttlZeroPod() *kapi.Pod {
```

Comment:
We are missing the test case for this pod. The pod should not be updated in this case.